components/monitoring/base/prometheus-servicemonitors.yaml

```diff
@@ -0,0 +1,46 @@
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  namespace: appstudio-workload-monitoring
+  name: prometheus
+  labels:
+    prometheus: appstudio-kcp-workload
+spec:
+  endpoints:
+  - bearerTokenSecret:
+      key: ""
+    interval: 15s
+    port: oauth2-proxy
+    scheme: HTTPS
+    tlsConfig:
+      caFile: /etc/prometheus/tls/tls.crt # volume mounted in the `prometheus` container of the prometheus pods
+      serverName: "prometheus-oauth2.appstudio-workload-monitoring.svc"
+  namespaceSelector:
+    matchNames:
+    - appstudio-workload-monitoring
+  selector:
+    matchLabels:
+      app.kubernetes.io/instance: prometheus
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  namespace: appstudio-workload-monitoring
+  name: spi-operator
+  labels:
+    prometheus: appstudio-kcp-workload
+spec:
+  endpoints:
+    - bearerTokenSecret:
+        name: prometheus-k8s-token-x6qjc
```

Comment:
where does this secret come from, and could we have a shorter name instead? Eg: `prometheus-k8s-token`

Reply:
This is the token used by the `prometheus-k8s` SA, I am not sure how to modify this secret name. But we should be able to decide this name on run time, I have a command for that. I am trying to figure out, how to do it using Argo.

Reply:
I thought we agreed that for now, all secrets could be generated by a script, which is why I wonder why you have such a name here

Reply:
this is not manually geerated. This is generated by prometheus operator/CR. so we dont need anything for generating this.